pkg/services/librarypanels/librarypanels_permissions_test.go

```diff
@@ -149,6 +149,23 @@ func TestLibraryPanelPermissions(t *testing.T) {
 				resp = sc.service.disconnectHandler(sc.reqContext)
 				require.Equal(t, testCase.status, resp.Status())
 			})
+
+		testScenario(t, fmt.Sprintf("When %s tries to delete all library panels in a folder with %s, it should return correct status", testCase.role, testCase.desc),
+			func(t *testing.T, sc scenarioContext) {
+				folder := createFolderWithACL(t, "Folder", sc.user, testCase.items)
+				cmd := getCreateCommand(folder.Id, "Library Panel Name")
+				resp := sc.service.createHandler(sc.reqContext, cmd)
+				validateAndUnMarshalResponse(t, resp)
+				sc.reqContext.SignedInUser.OrgRole = testCase.role
+
+				err := sc.service.DeleteLibraryPanelsInFolder(sc.reqContext, folder)
+				if testCase.status == 200 {
+					require.NoError(t, err)
+				}
+				if testCase.status == 403 {
+					require.EqualError(t, err, models.ErrFolderAccessDenied.Error())
+				}
```

Comment:
I think it might be better to use a switch, to ensure that all cases are covered, WDYT?

```suggestion
				switch testCase.status {
				case 200:
					require.NoError(t, err)
				case 403:
					require.EqualError(t, err, models.ErrFolderAccessDenied.Error())
				default:
					t.Fatalf("Unrecognized test case status %d", testCase.status)
				}
```

Reply:
I really like this, good catch!